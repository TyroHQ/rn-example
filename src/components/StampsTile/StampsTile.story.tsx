import { storiesOf } from "@storybook/react-native";
import figmaDecorator from "storybook-addon-figma";

import * as usage from "./StampsTile.usage";

let figma = undefined;

let story = storiesOf("StampsTile", module);

if (figma) {
  story = story.addDecorator(
    figmaDecorator({
      url: figma
    })
  );
}

for (const example in usage) {
  story = story.add(example, usage[example]);
}
