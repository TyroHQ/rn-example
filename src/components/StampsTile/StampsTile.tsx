import React from "react";
import { Text } from "react-native";
import styled from "styled-components/native";
import { Placeholder } from "../Placeholder";
import { RewardTileHeader } from "../RewardTileHeader";

const StampsTileContainer = styled.View`
  background-color: lightgrey;
  padding: 10px;
`;

export class StampsTile extends React.Component<{}> {
  static defaultProps: StampsTile["props"];
  render() {
    return (
      <Placeholder>
        Unlocked stamps open
        <RewardTileHeader />
        <Placeholder>
          White container
          <Placeholder>
            Radio boxes container
            <Placeholder>
              Radio boxes up
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>unselected radio box</Placeholder>
              <Placeholder>unselected radio box</Placeholder>
            </Placeholder>
            <Placeholder>
              Radio boxes down
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>selected radio box</Placeholder>
              <Placeholder>unselected radio box</Placeholder>
              <Placeholder>unselected radio box</Placeholder>
            </Placeholder>
          </Placeholder>
          <Placeholder>Get stamped button left</Placeholder>
          <Placeholder>Redeem button right</Placeholder>
        </Placeholder>
      </Placeholder>
    );
  }
}
