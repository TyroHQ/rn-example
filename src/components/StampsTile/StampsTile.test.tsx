import { shallow } from "enzyme";
import * as usage from "./StampsTile.usage";

describe("<StampsTile />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
