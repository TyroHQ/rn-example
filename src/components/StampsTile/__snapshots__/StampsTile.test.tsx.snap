// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StampsTile /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  Unlocked stamps open
  <RewardTileHeader />
  <Placeholder
    orientation="row"
  >
    White container
    <Placeholder
      orientation="row"
    >
      Radio boxes container
      <Placeholder
        orientation="row"
      >
        Radio boxes up
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          unselected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          unselected radio box
        </Placeholder>
      </Placeholder>
      <Placeholder
        orientation="row"
      >
        Radio boxes down
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          selected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          unselected radio box
        </Placeholder>
        <Placeholder
          orientation="row"
        >
          unselected radio box
        </Placeholder>
      </Placeholder>
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Get stamped button left
    </Placeholder>
    <Placeholder
      orientation="row"
    >
      Redeem button right
    </Placeholder>
  </Placeholder>
</Placeholder>
`;
